### DISCUSSION TITLE
Real time Data Sync over multiple browser/users

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#868] [Q&A] [answered]
Hi

_note: This could be a help to solve  #386 ,  #832 , #854_

I have 2 html pages, both with same app name and mv-storage source

[codepen1](https://codepen.io/galinhalx/pen/vYdwXeP)
[codepen2](https://codepen.io/galinhalx/pen/eYVadeE)

_btw, this could be same page, opened from different computers in fact._

Is there a way to see/refresh data inserted on one page, to the other page (and vice versa) ?

This feature works on @DmitrySharabin [chat app](https://dmitrysharabin.github.io/mavo-chat/), so ... it's possible :) where it updates sent messages on all browser windows 

but I don't understand how... :(

Is this because I should use Firebase storage?

Or is it because there's an additionnal [script](https://github.com/DmitrySharabin/mavo-chat/blob/main/index.html) ? 
(if so, is there a way to have something similar working for every app,
or some more help on how it works there)?
```

(async () => {
			await Mavo.ready;

			Mavo.DOMExpression.special.event("$user", {
				type: "mv-login mv-logout",
				update: (evt) => evt.backend?.user ?? {}
			});
		})();

		$.bind(document, "mv-load mv-save", () => {
			requestAnimationFrame(() => { // Give the browser a chance to render
				$("[mv-app=chat][mv-permissions~=logout] article[property=messages]:last-of-type")?.scrollIntoView(false);
			});
		});

		function save (name, editing = false) {
			const app = Mavo.all[name];

			if (editing && !app.unsavedChanges) {
				return;
			}

			app.save();
```

**Comment by DmitrySharabin** [comment, marked as answer]
> but I don't understand how... :(
> Is this because I should use Firebase storage?

Yup. Real-time updates is one of the features of the Firebase Firestore plugin: mind `mv-storage-options="realtime"` in the app's root.

> Or is it because there's an additionnal [script](https://github.com/DmitrySharabin/mavo-chat/blob/main/index.html) ?

I needed that because of the features we don't have in Mavo yet but that are good to have, and they were crucial for the Chat app.

**Reply by GalinhaLX** [reply]
does it mean it only needs to be connected with Firebase, and apps will update their content ?

or does it also need some additional javascript?

(sorry...just double checking if I fully understand your answer)

**Reply by DmitrySharabin** [reply]
> does it mean it only needs to be connected with Firebase, and apps will update their content ?

Right. Connect to Firebase and enable the `realtime` option. That's all. No need for other scripts.

> (sorry...just double checking if I fully understand your answer)

No worries! I'll always help if I can.
